src/main/java/seedu/uninurse/model/medication/MedicationList.java

```diff
@@ -0,0 +1,128 @@
+
+    /**
+     * Constructs an empty {@code MedicationList}.
+     */
+    public MedicationList() {
+        this.internalMedicationList = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a {@code MedicationList}.
+     * @param medicationList The given list of medications.
+     */
+    public MedicationList(List<Medication> medicationList) {
+        requireNonNull(medicationList);
+        this.internalMedicationList = medicationList;
+    }
+
+    @Override
+    public MedicationList add(Medication medication) {
+        requireNonNull(medication);
+        if (this.internalMedicationList.contains(medication)) {
+            throw new DuplicateMedicationException();
+        }
+
+        List<Medication> updatedMedications = new ArrayList<>(internalMedicationList);
+        updatedMedications.add(medication);
+        return new MedicationList(updatedMedications);
+    }
+
+    @Override
+    public MedicationList delete(int index) {
+        try {
+            List<Medication> updatedMedications = new ArrayList<>(internalMedicationList);
+            updatedMedications.remove(index);
+            return new MedicationList(updatedMedications);
+        } catch (IndexOutOfBoundsException e) {
+            throw new MedicationNotFoundException();
+        }
+    }
+
+    @Override
+    public MedicationList edit(int index, Medication editedMedication) {
+        try {
+            List<Medication> updatedMedications = new ArrayList<>(internalMedicationList);
+            if (updatedMedications.get(index).equals(editedMedication)) {
+                throw new UnmodifiedMedicationException();
+            }
+            updatedMedications.set(index, editedMedication);
+            return new MedicationList(updatedMedications);
+        } catch (IndexOutOfBoundsException e) {
+            throw new MedicationNotFoundException();
+        }
+    }
+
+    @Override
+    public Medication get(int index) {
+        try {
+            return internalMedicationList.get(index);
+        } catch (IndexOutOfBoundsException e) {
+            throw new MedicationNotFoundException();
+        }
+    }
+
+    @Override
+    public int size() {
+        return internalMedicationList.size();
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return internalMedicationList.isEmpty();
+    }
+
+    @Override
+    public List<Medication> getInternalList() {
+        return Collections.unmodifiableList(internalMedicationList);
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        internalMedicationList.forEach(m -> {
+            int index = internalMedicationList.indexOf(m);
+            if (index == 0) {
+                builder.append(index + 1)
+                        .append(". ")
+                        .append(m);
+            } else {
+                builder.append("\n")
+                        .append(index + 1)
```

Comment:
oops also a bug I found in my own branch
```suggestion
                        .append(index + 1)
                        .append(". ")
```

Reply:
Yup I amended it.